dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/liberty/CDIRuntimeImpl.java

```diff
@@ -467,8 +467,8 @@ public void applicationStarting(ApplicationInfo appInfo) throws StateChangeExcep
 
             ClassLoader appCL = getRealAppClassLoader(application);
             if (appCL != null) {
-                ClassLoader newCL = classLoadingSRRef.getServiceWithException().createThreadContextClassLoader(appCL);
-                application.setTCCL(newCL);
+                ClassLoader newCDISpecificCL = classLoadingSRRef.getServiceWithException().createThreadContextClassLoader(appCL);
+                application.setTCCL(newCDISpecificCL);
```

Comment:
I'm unclear about why this change is made, what's CDISpecific about this classloader?

Reply:
This classloader is created on line 470 and is only ever set onto the thread context when CDI methods are running.

(I made the change because David Zavala thought for a moment that CDI was destroying the main application classloader)

Reply:
Unfortunately, I don't think that's how it actually works. Although the method is called `createThreadContextClassLoader` it actually returns the same TCCL if passed the same app classloader, and increments a reference count (see [`CreateClassLoadingServiceImpl`](https://github.com/OpenLiberty/open-liberty/blob/c36e2355109d7838e558d2da9fb78b615b6dc35a/dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ClassLoadingServiceImpl.java#L650)).

Having the methods named `create` and `destroy` is misleading, but the TCCL we're getting back is not CDI specific.

That's actually important since libraries may use the TCCL as a key to a map (as I think MP Config does).

Reply:
It looks like you're correct, but the old name still shocked and confused. I'll ask David what he suggests we call it.

Reply:
Ideally we would change `ClassLoadingService` to have methods named `acquire` and `release` and document their effecnts, but it's SPI so we can't remove the existing methods. We could maybe improve the Javadoc.

I'm not sure we can improve anything by changing the variable name.

Reply:
I spoke to Dave and he recommended `appTCCL`